src/domain/mongoose_domain_api.erl

```diff
@@ -0,0 +1,130 @@
+%% Main module other parts of MongooseIM should use to access the domain
+%% management.
+-module(mongoose_domain_api).
+
+-export([init/1,
+         insert_domain/2,
+         remove_domain/1,
+         disable_domain/1,
+         enable_domain/1,
+         get_host_type/1,
+         get_all_locked/0,
+         get_domains_by_host_type/1]).
+
+-type domain() :: jid:lserver().
+-type host_type() :: jid:lserver().
+-type pair() :: {domain(), host_type()}.
+
+%% Init - on init all the “enabled” domain names from the persistent storage
+%% must be added to the core MIM component described above.
+%% Domains should be nameprepped using `jid:nameprep'
+-spec init([pair()]) -> ok | {error, term()}.
+init(Pairs) ->
```

Comment:
> Also host types are just strings. So every correct domain name is a correct host type identifier, but not every host type is a correct domain name

so, it would not make it interchangeable with any code, which does nameprepping for the `Host` argument. Or requires host-type in the specification (hook api-s, pool api-s).

Reply:
the host types automatically created for hosts declaration in config file will be interchangeable.

but the final goal is to ensure that MIM supports any string id for the host_type. that is one on the sanity checks.